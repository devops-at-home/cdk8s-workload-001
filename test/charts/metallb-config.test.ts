import { App } from 'cdk8s';
import { MetalLbConfig } from '../../src/charts/metallb-config.chart';

const chart = 'metallb-config';

describe(chart, () => {
    test('Snapshot test', () => {
        const app = new App();

        const manifest = new MetalLbConfig(app, chart, {
            namespace: 'kube-system',
            addresses: ['127.0.0.1/8'],
        });

        expect(manifest.toJson()).toMatchSnapshot();
    });
});
