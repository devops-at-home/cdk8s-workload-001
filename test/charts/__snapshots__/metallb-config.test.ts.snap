// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metallb-config Snapshot test 1`] = `
[
  {
    "apiVersion": "metallb.io/v1beta1",
    "kind": "IPAddressPool",
    "metadata": {
      "name": "metallb-config-ipaddresspool-c8fcbc5a",
      "namespace": "kube-system",
    },
    "spec": {
      "addresses": [
        "127.0.0.1/8",
      ],
    },
  },
  {
    "apiVersion": "metallb.io/v1beta1",
    "kind": "L2Advertisement",
    "metadata": {
      "name": "metallb-config-l2advertisement-c8dc6833",
      "namespace": "kube-system",
    },
    "spec": {
      "interfaces": [
        "tailscale0",
      ],
      "ipAddressPools": [
        "metallb-config-ipaddresspool-c8fcbc5a",
      ],
    },
  },
]
`;
